frontend/app/runner/selectvenue/page.tsx

```diff
@@ -0,0 +1,63 @@
+"use client";
+
+import { Button } from "@/components/ui/button"
+import { BackButton } from "@/components/backbutton"
+import { useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+
+export default function Home() {
+  const API_URL = process.env.NEXT_PUBLIC_API_URL;
+  const router = useRouter();
+
+  const [venues, setVenues] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const load = async () => {
+      try {
+        const res = await fetch(`${API_URL}/venues`);
+        if (!res.ok) throw new Error("Failed to fetch venues");
+        const data = await res.json();
+        setVenues(data);
+      } catch (err) {
+        console.error(err);
+      }
+      setLoading(false);
+    };
+
+    load();
+  }, [API_URL]);
+
+
+  return (
+    <main className="p-2">
+      <BackButton href="/runner" />
+
+      <div>
+        <h1 className="text-3xl font-bold">
+          Hey <span className="text-green-600">Runner</span>
+        </h1>
+        <h2 className="text-xl font-semibold mt-2">
+          Select your venue
+        </h2>
+
+        {loading && <p>Loading...</p>}
+
+        <ul>
+          {!loading && venues.length > 0 && venues.map((venue: any) => (
+            <li key={venue.id}>
+              <Button
+                className="bg-primary1 h-11 rounded-md"
+                onClick={() => router.push(`/runner/${venue.id}/selectstall`)}
```

Comment:
```suggestion
                onClick={() => router.push(`/runner/${venue.venue_id}/selectstall`)}
```
Again, with the venue_id attribute.